GitUI/Editor/FileViewer.cs

```diff
@@ -1539,24 +1539,43 @@ private void ApplySelectedLines(bool allFile, bool reverse)
             }
 
             byte[]? patch;
-            if (reverse)
+
+            Validates.NotNull(_viewItem);
+            if (_viewItem.Item.IsNew)
             {
-                patch = PatchManager.GetResetWorkTreeLinesAsPatch(
+                var treeGuid = reverse ? _viewItem.Item.TreeGuid?.ToString() : null;
+                patch = PatchManager.GetSelectedLinesAsNewPatch(
                     Module,
+                    _viewItem.Item.Name,
                     GetText(),
                     selectionStart,
                     selectionLength,
-                    Encoding);
+                    Encoding,
+                    reset: reverse,
+                    FilePreamble,
+                    treeGuid);
             }
             else
```

Comment:
reduce indentation like in ResetNoncommittedSelectedLines
```suggestion
            else if (reverse)
```